admiral/envs/components/agent.py

```diff
@@ -56,8 +63,14 @@ def __init__(self, observation_space=None, **kwargs):
         self.observation_space = {} if observation_space is None else observation_space
 
     def finalize(self, **kwargs):
+        """
+        Wrap all the observation spaces with a Dict and seed it if the agent was
+        created with a seed.
+        """
         super().finalize(**kwargs)
         self.observation_space = Dict(self.observation_space)
+        if self.seed is not None:
```

Comment:
Same as above: I don't think the 'if ...' is necessary. It makes sense to seed every time and the underlying function can handle 'None' values.